src/pages/purchase/kinguin.tsx

```diff
@@ -0,0 +1,216 @@
+    }
+
+    // Initialize checkout when SDK is loaded
+    if (sdkLoaded) {
+      initializeCheckout();
+    }
+  }, [productId, sdkLoaded]);
+
+  // Cleanup on unmount
+  useEffect(() => {
+    return () => {
+      if (window.kinguinCheckoutSDK && checkoutInitialized.current) {
+        try {
+          // No explicit destroy method mentioned in docs, but cleanup if needed
+          console.log('Cleaning up Kinguin SDK');
+        } catch (err) {
+          console.error('Error cleaning up Kinguin SDK:', err);
+        }
+      }
+    };
+  }, []);
+
+  if (!productId) {
+    return (
+      <Container size="sm" mt="xl">
+        <Alert icon={<IconAlertCircle size={16} />} title="Invalid Request" color="red">
+          Missing required product information. Please return to the gift cards page and try again.
+        </Alert>
+      </Container>
+    );
+  }
+
+  return (
+    <>
+      <Meta
+        title="Complete Your Purchase | Civitai"
+        description="Complete your gift card purchase securely through Kinguin"
+      />
+
+      <Script
+        src="https://static.kinguin.net/checkout/sdk/sdk-1.2.0.min.js"
+        onLoad={handleSDKLoad}
+        onError={() => setError('Failed to load checkout SDK. Please try again.')}
+      />
+
+      <div className="relative -mt-4 size-full min-h-[700px]">
+        {loading && (
+          <Center className="absolute inset-0 z-10">
+            <div className="text-center">
+              <Loader size="lg" />
+              <Text mt="md">Loading checkout...</Text>
+              {productName && (
+                <Text size="sm" c="dimmed" mt="xs">
+                  {productName}
+                </Text>
+              )}
+            </div>
+          </Center>
+        )}
+
+        {error && (
+          <Container size="sm" mt="xl">
+            <Alert icon={<IconAlertCircle size={16} />} title="Checkout Error" color="red">
+              {error}
+            </Alert>
+          </Container>
+        )}
+
+        {/* Kinguin SDK wrapper - this is where the iframe will be injected */}
+        <div id="kinguin-checkout-wrapper" className="relative size-full min-h-[700px]">
+          {/* Product button that triggers the checkout */}
+          <button
+            className="kinguin-product-button absolute inset-0 size-full cursor-default opacity-0"
+            data-product-url={productId ? buildKinguinUrl(productId) : ''}
+            style={{ zIndex: -1 }}
+          >
+            {/* Hidden trigger button for SDK */}
+          </button>
+        </div>
+      </div>
+    </>
+  );
+}
+
+export const getServerSideProps = createServerSideProps({
+  useSSG: true,
+  useSession: true,
+  resolver: async ({ session }) => {
+    if (!session?.user || session.user.bannedAt) {
+      return {
+        redirect: {
+          destination: '/login?returnUrl=/purchase/kinguin',
+          permanent: false,
+        },
+      };
+    }
+
+    return {
+      props: {},
+    };
+  },
+});
```

Comment:
Just a little nitpick, but I'd put this at the top of the file just to keep consistency, also we don't need useSSG here I believe